### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by bmukha**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,36 +1,89 @@
...
+						element={token ? <Home token={token} /> : <Registration />}
+					/>
+					<Route
+						path='login'
+						element={
+							token ? (
+								<Home token={token} />
+							) : (
+								<Login token={token} setName={setName} setToken={setToken} />
+							)
```
Comment (EpamFanAnton) [inline src/App.tsx]: Don't really like your approach with `Home` component.
You just should add a check inside your pages that redirects user where needed

Comment (EpamFanAnton) [reply src/App.tsx]: For example:
```
const Component = () => {
  const navigate = useNavigate()
  const token = useSelector(tokenSelector)
  
  useEffect(() => {
    !token && navigate('/login')
  }, [ token ])
}
```

File: src/components/Registration/Registration.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+		target: { value },
+	}): void => setName(value);
+
+	const handleEmailChange: ChangeEventHandler<HTMLInputElement> = ({
+		target: { value },
+	}): void => setEmail(value);
+
+	const handlePasswordChange: ChangeEventHandler<HTMLInputElement> = ({
+		target: { value },
+	}): void => setPassword(value);
```
Comment (EpamFanAnton) [inline src/components/Registration/Registration.tsx]: these handlers are redundant. You can do it in your `onChange` listener. `onChange={ ({ target: { value } }) => setName(value) }`

**Review by EpamFanAnton — CHANGES_REQUESTED** [review]

**Comment by bmukha** [comment]
@EpamFanAnton I've implemented all proposed changes, please review.

**Review by EpamFanAnton — APPROVED** [review]
25

**Pull request merged**